Quickbuild - "Invalid time value" - error

### Discussed in https://github.com/Accenture/sfpowerscripts/discussions/696

<div type='discussions-op-text'>

<sup>Originally posted by **bjanderson70** September 21, 2021</sup>
**Problem Statement:**
I have a github action that fires when I check-in/commit. There appears to be an invalid time-value passed to _getFormattedTime_ in **GetFormattedTime.ts**

**_Suggestions_**? Short of putting a try/catch around the code ...I have not ventured into the sfdx cli to determine what cause this ... looking for some suggestions in the meantime? TIA

**Command:** - ran in a github action
_sfdx sfpowerscripts:orchestrator:quickbuild -v devhub --diffcheck --buildnumber ${GITHUB_RUN_ID} --branch ${GITHUB_REF#refs/heads/} --json --waittime=60_

**Error Encountered:**


Run sfdx sfpowerscripts:orchestrator:quickbuild -v devhub --diffcheck --buildnumber ${GITHUB_RUN_ID} --branch ${GITHUB_REF#refs/heads/} --json --waittime=60
 ›   Warning: sfdx-cli update available from 7.110.0 to 7.116.2.
-----------sfpowerscripts orchestrator ------------------
command: quickbuild
Build Packages Only Changed: true
Config File Path: config/project-scratch-def.json
Artifact Directory: artifacts
---------------------------------------------------------

Checking last known tags for cqrs-service to determine whether package is to be built...

Utilizing tag cqrs-service_v1.0.0.1251293990 for cqrs-service
Checking for changes in source directory src/utils/cqrs
Checking for changes to package descriptor in sfdx-project.json
Packages scheduled for build
┌─────────┬────────────────────┬────────────────┐
│ Package │ Reason to be built │ Last Known Tag │
└─────────┴────────────────────┴────────────────┘
No packages found to be built.. Exiting.. 
----------------------------------------------------------------------------------------------------
{
    **"status": 1,**
    "name": "RangeError",
    "message": "Invalid time value",
    "exitCode": 1,
    "commandName": "QuickBuild",
    "stack": "RangeError: Invalid time value\n    at Date.toISOString (<anonymous>)\n    at Object.getFormattedTime [as default] (/sfdx_plugins/.local/share/sfdx/node_modules/@dxatscale/sfpowerscripts/lib/utils/GetFormattedTime.js:6:27)\n    at QuickBuild.<anonymous> (/sfdx_plugins/.local/share/sfdx/node_modules/@dxatscale/sfpowerscripts/lib/BuildBase.js:89:171)\n    at Generator.next (<anonymous>)\n    at fulfilled (/sfdx_plugins/.local/share/sfdx/node_modules/@dxatscale/sfpowerscripts/lib/BuildBase.js:5:58)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\nOuter stack:\n    at Function.wrap (/sfdx_plugins/.local/share/sfdx/node_modules/@salesforce/core/lib/sfdxError.js:171:27)\n    at QuickBuild.catch (/sfdx_plugins/.local/share/sfdx/node_modules/@dxatscale/sfpowerscripts/node_modules/@salesforce/command/lib/sfdxCommand.js:255:67)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at async QuickBuild._run (/sfdx_plugins/.local/share/sfdx/node_modules/@dxatscale/sfpowerscripts/node_modules/@salesforce/command/lib/sfdxCommand.js:95:13)\n    at async Config.runCommand (/usr/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)\n    at async SfdxMain.run (/usr/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)\n    at async SfdxMain._run (/usr/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)\n    at async Object.run (/usr/lib/node_modules/sfdx-cli/dist/cli.js:162:47)",
    "warnings": []
}
Error: Process completed with exit code 1.
</div>

--- a/packages/sfpowerscripts-cli/src/BuildBase.ts
+++ b/packages/sfpowerscripts-cli/src/BuildBase.ts
@@ -9,7 +9,13 @@ import SFPStatsSender from "@dxatscale/sfpowerscripts.core/lib/stats/SFPStatsSen
 import BuildImpl from "./impl/parallelBuilder/BuildImpl";
 import { Stage } from "./impl/Stage";
 import PackageMetadata from "@dxatscale/sfpowerscripts.core/lib/PackageMetadata";
-import { COLOR_ERROR, COLOR_HEADER,COLOR_INFO,COLOR_TIME,COLOR_SUCCESS } from "@dxatscale/sfpowerscripts.core/lib/logger/SFPLogger"
+import {
+  COLOR_ERROR,
+  COLOR_HEADER,
+  COLOR_INFO,
+  COLOR_TIME,
+  COLOR_SUCCESS,
+} from "@dxatscale/sfpowerscripts.core/lib/logger/SFPLogger";
 import getFormattedTime from "./utils/GetFormattedTime";
 
 // Initialize Messages with the current plugin directory
@@ -32,7 +38,10 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
     gittag: flags.boolean({
       description: messages.getMessage("gitTagFlagDescription"),
       hidden: true,
-      deprecated: {messageOverride:"--gittag is deprecated, Please utilize git tags on publish stage"}
+      deprecated: {
+        messageOverride:
+          "--gittag is deprecated, Please utilize git tags on publish stage",
+      },
     }),
     repourl: flags.string({
       char: "r",
@@ -61,7 +70,7 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
     }),
     branch: flags.string({
       description: messages.getMessage("branchFlagDescription"),
-      required: true
+      required: true,
     }),
     tag: flags.string({
       description: messages.getMessage("tagFlagDescription"),
@@ -89,31 +98,43 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
         "ERROR",
         "FATAL",
       ],
-    })
+    }),
   };
 
   public async execute() {
-    let buildExecResult: {generatedPackages: PackageMetadata[], failedPackages: string[]};
+    let buildExecResult: {
+      generatedPackages: PackageMetadata[];
+      failedPackages: string[];
+    };
     let totalElapsedTime: number;
     let artifactCreationErrors: string[] = [];
 
     let tags;
     try {
-
-    
-     
-
-
       const artifactDirectory: string = this.flags.artifactdir;
       const diffcheck: boolean = this.flags.diffcheck;
       const branch: string = this.flags.branch;
 
-      console.log(COLOR_HEADER("-----------sfpowerscripts orchestrator ------------------"));
+      console.log(
+        COLOR_HEADER(
+          "-----------sfpowerscripts orchestrator ------------------"
+        )
+      );
       console.log(COLOR_HEADER(`command: ${this.getStage()}`));
-      console.log(COLOR_HEADER(`Build Packages Only Changed: ${this.flags.diffcheck}`));
-      console.log(COLOR_HEADER(`Config File Path: ${this.flags.configfilepath}`));
-      console.log(COLOR_HEADER(`Artifact Directory: ${this.flags.artifactdir}`));
-      console.log(COLOR_HEADER("---------------------------------------------------------"));
+      console.log(
+        COLOR_HEADER(`Build Packages Only Changed: ${this.flags.diffcheck}`)
+      );
+      console.log(
+        COLOR_HEADER(`Config File Path: ${this.flags.configfilepath}`)
+      );
+      console.log(
+        COLOR_HEADER(`Artifact Directory: ${this.flags.artifactdir}`)
+      );
+      console.log(
+        COLOR_HEADER(
+          "---------------------------------------------------------"
+        )
+      );
 
       let executionStartTime = Date.now();
 
@@ -138,9 +159,13 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
       ) {
         console.log(`${EOL}${EOL}`);
         console.log(COLOR_INFO("No packages found to be built.. Exiting.. "));
-        return;
+        console.log(
+          COLOR_HEADER(
+            `----------------------------------------------------------------------------------------------------`
+          )
+        );
+        throw new Error("No packages to be found to be built")
       }
-
 
       console.log(`${EOL}${EOL}`);
       console.log("Generating Artifacts and Tags....");
@@ -153,7 +178,6 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
             artifactDirectory,
             generatedPackage
           );
-
         } catch (error) {
           console.log(error.message);
           artifactCreationErrors.push(generatedPackage.package_name);
@@ -162,92 +186,109 @@ export default abstract class BuildBase extends SfpowerscriptsCommand {
 
       totalElapsedTime = Date.now() - executionStartTime;
 
-      if (artifactCreationErrors.length > 0 || buildExecResult.failedPackages.length > 0)
+      if (
+        artifactCreationErrors.length > 0 ||
+        buildExecResult.failedPackages.length > 0
+      )
         throw new Error("Build Failed");
 
-
-      
-  
       SFPStatsSender.logGauge(
         "build.duration",
         Date.now() - executionStartTime,
         tags
       );
 
-      SFPStatsSender.logCount(
-        "build.succeeded",
-        tags
-      );
-
+      SFPStatsSender.logCount("build.succeeded", tags);
     } catch (error) {
-      SFPStatsSender.logCount(
-        "build.failed",
-        tags
-      );
+      SFPStatsSender.logCount("build.failed", tags);
       console.log(COLOR_ERROR(error));
       process.exitCode = 1;
     } finally {
-      console.log(COLOR_HEADER(
-        `----------------------------------------------------------------------------------------------------`
-      ));
-      console.log(COLOR_SUCCESS(
-        `${buildExecResult.generatedPackages.length} packages created in ${COLOR_TIME(getFormattedTime(totalElapsedTime))} minutes with {${COLOR_ERROR(buildExecResult.failedPackages.length)}} errors}`
-      ));
+      if (
+        buildExecResult?.generatedPackages?.length > 0 ||
+        buildExecResult?.failedPackages?.length > 0
+      ) {
+        console.log(
+          COLOR_HEADER(
+            `----------------------------------------------------------------------------------------------------`
+          )
+        );
+        console.log(
+          COLOR_SUCCESS(
+            `${
+              buildExecResult.generatedPackages.length
+            } packages created in ${COLOR_TIME(
+              getFormattedTime(totalElapsedTime)
+            )} minutes with {${COLOR_ERROR(
+              buildExecResult.failedPackages.length
+            )}} errors}`
+          )
+        );
 
-      if (buildExecResult.failedPackages.length > 0)
-        console.log(COLOR_ERROR(`Packages Failed To Build`, buildExecResult.failedPackages));
+        if (buildExecResult.failedPackages.length > 0)
+          console.log(
+            COLOR_ERROR(
+              `Packages Failed To Build`,
+              buildExecResult.failedPackages
+            )
+          );
 
+        if (artifactCreationErrors.length > 0)
+          console.log(
+            COLOR_ERROR(`Failed To Create Artifacts`, artifactCreationErrors)
+          );
 
-      if (artifactCreationErrors.length > 0)
-        console.log(COLOR_ERROR(`Failed To Create Artifacts`, artifactCreationErrors));
+        console.log(
+          COLOR_HEADER(
+            `----------------------------------------------------------------------------------------------------`
+          )
+        );
 
-      console.log(
-        COLOR_HEADER(`----------------------------------------------------------------------------------------------------`
-      ));
+        const buildResult: BuildResult = {
+          packages: [],
+          summary: {
+            scheduled_packages: null,
+            elapsed_time: null,
+            succeeded: null,
+            failed: null,
+          },
+        };
 
-      const buildResult: BuildResult = {
-        packages: [],
-        summary: {
-          scheduled_packages: null,
-          elapsed_time: null,
-          succeeded: null,
-          failed: null,
+        for (let generatedPackage of buildExecResult.generatedPackages) {
+          buildResult["packages"].push({
+            name: generatedPackage["package_name"],
+            version: generatedPackage["package_version_number"],
+            elapsed_time: generatedPackage["creation_details"]?.creation_time,
+            status: "succeeded",
+          });
         }
-      };
 
-      for (let generatedPackage of buildExecResult.generatedPackages) {
-        buildResult["packages"].push({
-          name: generatedPackage["package_name"],
-          version: generatedPackage["package_version_number"],
-          elapsed_time: generatedPackage["creation_details"]?.creation_time,
-          status: "succeeded",
-        });
+        for (let failedPackage of buildExecResult.failedPackages) {
+          buildResult["packages"].push({
+            name: failedPackage,
+            version: null,
+            elapsed_time: null,
+            status: "failed",
+          });
+        }
+
+        buildResult["summary"].scheduled_packages =
+          buildExecResult.generatedPackages.length +
+          buildExecResult.failedPackages.length;
+        buildResult["summary"].elapsed_time = totalElapsedTime;
+        buildResult["summary"].succeeded =
+          buildExecResult.generatedPackages.length;
+        buildResult["summary"].failed = buildExecResult.failedPackages.length;
+
+        fs.writeFileSync(
+          `buildResult.json`,
+          JSON.stringify(buildResult, null, 4)
+        );
       }
-
-      for (let failedPackage of buildExecResult.failedPackages) {
-        buildResult["packages"].push({
-          name: failedPackage,
-          version: null,
-          elapsed_time: null,
-          status: "failed",
-        });
-      }
-
-      buildResult["summary"].scheduled_packages =
-        buildExecResult.generatedPackages.length + buildExecResult.failedPackages.length;
-      buildResult["summary"].elapsed_time = totalElapsedTime;
-      buildResult["summary"].succeeded = buildExecResult.generatedPackages.length;
-      buildResult["summary"].failed = buildExecResult.failedPackages.length;
-
-      fs.writeFileSync(
-        `buildResult.json`,
-        JSON.stringify(buildResult, null, 4)
-      );
     }
   }
 
   abstract getStage(): Stage;
-
 
   abstract getBuildImplementer(): BuildImpl;
 }
@@ -266,5 +307,3 @@ interface BuildResult {
     failed: number;
   };
 }
-
-

